packages/shared/components/popups/SendFormPopup.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="typescript">
+    import { Button, Text, RecipientInput, AssetAmountInput } from 'shared/components'
+    import { clearSendParams } from 'shared/lib/app'
+    import { localize } from '@core/i18n'
+    import { closePopup, openPopup } from 'shared/lib/popup'
+    import { FontWeightText } from 'shared/components/Text.svelte'
+    import { UNIT_MAP } from '@lib/units'
+
+    let asset
+    let amount
+    let unit
+    let recipient
+
+    let assetAmountInput
+    let recipientInput
+
+    async function onSend() {
+        let valid = true
+
+        async function validate() {
+            await Promise.allSettled([
+                assetAmountInput?.validate().then(
+                    () => {},
+                    () => {
+                        valid = false
+                    }
+                ),
+                recipientInput?.validate().then(
+                    () => {},
+                    () => {
+                        valid = false
+                    }
+                ),
+            ])
```

Comment:
This is because we want to run all validation functions on submit even if one fails, so that we get the correct error messages to display for each input in the form. But also if there is a failure we want to set valid to false.

Reply:
Okay, this is interesting, thanks for clarifying!